### PR TITLE
Added Exponent Search and updated README.md

### PR DESCRIPTION
**Pull request by Sarthak-9**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

### Description

Added Exponential Search in C++

### Checklist

- [# ] I've read the contribution guidelines.
- [#] I've checked the issue list before deciding what to submit.
- [# ] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

#1826

### CODE REVIEW

**Comment by Sarthak-9** [comment]
@shraddhavp @dwoc Any updates

File: C-Plus-Plus/search/exponential_search.cpp [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+        // repeated doubling method to find range for binary search
+        int i = 1;
+        while (i < n && arr[i] <= key) {
+                i = i * 2;
+        }
+
+        int mini = i < (n-1) ? i : (n-1);
+
+        // BinarySearch called
+        return BinarySearch(arr, i/2, mini, key);
```
Comment (atarax665) [inline C-Plus-Plus/search/exponential_search.cpp]:
```suggestion
        return BinarySearch(arr, i / 2, mini, key);
```

File: C-Plus-Plus/search/exponential_search.cpp [hunk 2]
```diff
@@ -0,0 +1,97 @@
+/*
+ Created by Sarthak-9
```
Comment (atarax665) [inline C-Plus-Plus/search/exponential_search.cpp:3]: Add more description about Exponential search. i.e - How it works and in what cases it should be preferred. Also mention the space and time complexity of the algorithm.

File: C-Plus-Plus/search/exponential_search.cpp [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+        }
+
+        int key;
+        cout<<"Enter the element to search"<<endl;
+        cin >> key;
+
+        int search_result = ExponentialSearch(arr, n, key);
+
+        if (search_result != -1) {
+                cout << "Element is present at position " << search_result+1 << endl;
```
Comment (atarax665) [inline C-Plus-Plus/search/exponential_search.cpp]: Output should give the index at which the element is present according to the general convention.
```suggestion
                cout << "Element is present at position " << search_result << endl;
```

File: C-Plus-Plus/search/exponential_search.cpp [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+ Input:
+ Enter the size of array
+ 8
+ Enter the sorted array
+ 5 12 25 38 52 76 110 155
+ Enter the element to search
+ 52
+
+ Output:
+ Element is present at position 5
```
Comment (atarax665) [inline C-Plus-Plus/search/exponential_search.cpp]:
```suggestion
 Element is present at position 4
```

File: C-Plus-Plus/search/exponential_search.cpp [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+
+*/
```
Comment (atarax665) [inline C-Plus-Plus/search/exponential_search.cpp]: Add a line break at the end of the program.

**Review by atarax665 — CHANGES_REQUESTED** [review]

**Comment by Sarthak-9** [comment]
@atarax665 Check now

**Pull request merged**